From 04b118acce2a74306f365ef329cbe00efc399b26 Mon Sep 17 00:00:00 2001
From: Denis Chenu <denis@sondages.pro>
Date: Mon, 23 Mar 2020 18:30:55 +0100
Subject: [PATCH] Fixed issue [security] #16019: Stored XSS in survey groups.
 (Thanks to Matthew Aberegg and Michael Burkey) Dev: admin view to be fixed
 Dev: not shown in public : no need XSS filtering

---
 application/models/SurveysGroups.php                     | 1 -
 application/views/admin/surveysgroups/surveySettings.php | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/application/models/SurveysGroups.php b/application/models/SurveysGroups.php
index cc20fe668a..65c2b2b496 100644
--- a/application/models/SurveysGroups.php
+++ b/application/models/SurveysGroups.php
@@ -41,7 +41,6 @@ public function rules()
             array('name', 'match', 'pattern'=> '/^[A-Za-z0-9_\.]+$/u','message'=> gT('Group name can contain only alphanumeric character, underscore or dot.')),
             array('title', 'length', 'max'=>100),
             array('description, created, modified', 'safe'),
-            array('title, name, description', 'LSYii_Validators'),
             // The following rule is used by search().
             // @todo Please remove those attributes that should not be searched.
             array('gsid, name, title, description, owner_id, parent_id, created, modified, created_by', 'safe', 'on'=>'search'),
diff --git a/application/views/admin/surveysgroups/surveySettings.php b/application/views/admin/surveysgroups/surveySettings.php
index 2d41c244fd..024b114198 100644
--- a/application/views/admin/surveysgroups/surveySettings.php
+++ b/application/views/admin/surveysgroups/surveySettings.php
@@ -31,7 +31,7 @@
                 )
                 )
             )); ?>
-    <h3><?php eT('Survey settings for group: '); echo '<strong><em>'.$model->title.'</strong></em>'; ?></h3>
+    <h3><?php eT('Survey settings for group: '); echo '<strong><em>'.CHtml::encode($model->title).'</strong></em>'; ?></h3>
     <div class="row">
         <div id="surveySettingsForThisGroup" style="display: flex; flex-wrap:nowrap;">
             <div id="global-sidebar-container">
